NeoML/test/src/ReferenceDnnTest.cpp

```diff
@@ -0,0 +1,279 @@
+/* Copyright @ 2024 ABBYY
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#include <common.h>
+#include <mutex>
+#pragma hdrstop
+
+#include <TestFixture.h>
+
+using namespace NeoML;
+using namespace NeoMLTest;
+
+namespace NeoMLTest {
+
+struct CDnnTestParam {
+	CDnn* net;
+};
+
+static IThreadPool::TFunction runDnn = [](int, void* params)
+{
+	CDnnTestParam* taskParams = static_cast<CDnnTestParam*>(params);
+	taskParams->net->RunOnce();
+};
+
+static CDnn* createDnn(CRandom& random)
+{
+	CDnn* net = new CDnn(random, MathEngine());
+
+	CPtr<CSourceLayer> dataLayer = Source(*net, "in");
+
+	CPtr<CFullyConnectedLayer> fc1 = new CFullyConnectedLayer(MathEngine());
```

Comment:
_Minor fix_
Please, rewrite using functional interface
```
FullyConnected( 50, false )(dataLayer);
CDropoutLayer(0.1f)(MathEngine());
``
its need less code, and it may find some problems in scenarios used by users
(I already missed 1 bug with it :( )